File: backend/app/schemas/appointment.py
"""Appointment schemas — request/response for booking endpoints."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.core.enums import AppointmentStatus


class AppointmentCreate(BaseModel):
    """Schema for booking an appointment."""
    slot_id: uuid.UUID
    reason: str | None = Field(None, max_length=500)


class AppointmentCancel(BaseModel):
    """Schema for cancelling an appointment."""
    cancellation_reason: str | None = Field(None, max_length=500)


class AppointmentResponse(BaseModel):
    """Appointment data returned in API responses."""
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    patient_id: uuid.UUID
    doctor_id: uuid.UUID
    slot_id: uuid.UUID
    status: AppointmentStatus
    reason: str | None
    notes: str | None
    cancelled_at: datetime | None
    cancellation_reason: str | None
    created_at: datetime


class AppointmentDetailResponse(AppointmentResponse):
    """Extended response with joined data for detail views."""
    doctor_name: str | None = None
    patient_name: str | None = None
    slot_date: str | None = None
    slot_time: str | None = None

File: frontend/lib/models/appointment.dart
class Appointment {
  final String id;
  final String patientId;
  final String doctorId;
  final String slotId;
  final String status;
  final String? reason;
  
  // Joined fields for convenience
  final String? doctorName;
  final String? patientName;
  final String? slotDate;
  final String? slotTime;

  Appointment({
    required this.id,
    required this.patientId,
    required this.doctorId,
    required this.slotId,
    required this.status,
    this.reason,
    this.doctorName,
    this.patientName,
    this.slotDate,
    this.slotTime,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'] as String,
      patientId: json['patient_id'] as String,
      doctorId: json['doctor_id'] as String,
      slotId: json['slot_id'] as String,
      status: json['status'] as String,
      reason: json['reason'] as String?,
      doctorName: json['doctor_name'] as String?,
      patientName: json['patient_name'] as String?,
      slotDate: json['slot_date'] as String?,
      slotTime: json['slot_time'] as String?,
    );
  }
}
